brave-tests/src/main/java/brave/test/ITRemote.java

```diff
@@ -239,15 +239,39 @@ private Span doTakeSpan(boolean flushed) throws InterruptedException {
     }
   };
 
+  /**
+   * Blocks until a local span was reported. We define a local span as one with a timestamp and no
+   * duration, kind, or remote endpoint. This will fail if there's an "error" tag. If you expect a
+   * failure, use {@link #takeLocalSpanWithError(String)} instead.
+   */
   protected Span takeLocalSpan() throws InterruptedException {
     Span local = takeSpan();
+    assertLocalSpan(local);
+    return local;
+  }
+
+  /** Like {@link #takeLocalSpan()} except an error tag must match the given value. */
+  protected Span takeLocalSpanWithError(String errorTag)
```

Comment:
I wonder if this results in too many permutations. I'm thinking of adding `BraveAssertions` to allow something like

```java
assertThat(takeSpan()).isLocal().hasError("failed");
```

Not something you have to do :) But wondering what you think of the idea.

Reply:
right now there are two permutations, local and remote. there can't be any other permutations.

the explicit handling of error is really really important because until we made error mandatory we actually had bugs we never noticed.

Reply:
it could be the case that some of the comparison operators could fit in an assertions class though, ex childOf parentOf etc

let's just please not take the base preconditions out, that verify this is or is not a local span and is or is not error. deferring these things would re-introduce the problems the code just only for a day has so far solved.

Reply:
Ah ok makes sense - remembered that `takeSpan` verifies there is no error which does seem better.

Reply:
the other thing is accidental permutation. we had tests that thought they were acting on a local span when they were acting on a remote one.. I tried to limit most of this by using scopes instead of parent spans